src/components/listings/ImageGrid.tsx

```diff
@@ -0,0 +1,28 @@
+import { ImageGridProps } from '@/types/imageGrid'
+import React, { useEffect, useState } from 'react'
+import { View } from 'react-native'
+import ImageGridItem from '@/components/listings/ImageGridItem'
+
+function ImageGrid(props: ImageGridProps) {
+  const { images } = props
+
+  const [selectedImages, setSelectedImages] = useState<Array<number>>([])
+
+  useEffect(() => {
+    console.log(selectedImages)
```

Comment:
Shall we set up a `handleSelectedImages` function that also emits the result via props `onSelectedImage`?

Reply:
👍

Reply:
What i meant is you need a function called `handleSelectedImages` that will set the internal state and pass the state outside to the parent.

Then we don't need this useEffect :)